packages/kbn-unified-field-list/src/hooks/use_existing_fields.ts

```diff
@@ -251,7 +254,12 @@ export const useExistingFieldsFetcher = (
   );
 };
 
-export const useExistingFieldsReader: () => ExistingFieldsReader = () => {
+export const useExistingFieldsReader: () => {
+  getFieldsExistenceStatus: (dataViewId: string) => ExistenceFetchStatus;
+  isFieldsExistenceInfoUnavailable: (dataViewId: string) => boolean;
+  getNewFields: (dataViewId: string) => FieldSpec[];
+  hasFieldData: (dataViewId: string, fieldName: string) => boolean;
```

Comment:
I guess we can keep `ExistingFieldsReader` here.

Reply:
Definitely! Thx